buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -590,6 +589,16 @@ class BuildPlugin implements Plugin<Project> {
                 url "https://s3.amazonaws.com/download.elasticsearch.org/lucenesnapshots/${revision}"
             }
         }
+        String compilerJavaHome = findCompilerJavaHome()
+        String runtimeJavaHome = findRuntimeJavaHome(compilerJavaHome)
+        if (runtimeJavaHome.contains("fips")) {
```

Comment:
Could we take this opportunity to clean things up and have something like `-Dfips.enabled=true` or `-Dtests/fips=true` rather than rely on the runtime java home name ? 
My understanding is that we no longer need to make modifications to that location, so it will only really act as a flag.

Reply:
Yes, good idea ! What we would need to do is to bundle the `java.policy` and `java.security` (one for each jdk version) as test resources and adjust the system properties (` -Djava.security.policy` and ` -Djava.security.properties` IIRC) the same way we do for the keystore password. Would that be ok ?

Reply:
I think that's better as it will have all the fips config contained in our build. 
You'll probably want to look at the `ExportElasticsearchBuildResourcesTask` tasks , there's already one created for each project.